src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -529,11 +514,6 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
         ProgrammingExercise programmingExercise = programmingExerciseService.findByIdWithEagerStudentParticipationsAndSubmissions(exerciseId);
         User user = userService.getUserWithGroupsAndAuthorities();
 
-        if (Optional.ofNullable(programmingExercise).isEmpty()) {
```

Comment:
Not needed since the `findByIdWIth....` method already throws an exception in this case